tools/pre-commit

```diff
@@ -1 +1,22 @@
+#!/bin/bash
+
+# This hook runs the Zulip code linter ./tools/lint-all and returns true
+# regardless of linter results so that your commit may continue.
+
+# Messages from the linter will output to screen.
+#
+# If you are using the vagrant method for your Zulip dev setup, the linter will
+# be run through vagrant ssh.
+
+if ! [[ "$VIRTUAL_ENV" == "/srv/zulip-venv" ]] && `which vagrant > /dev/null` && [ -e .vagrant ]
```

Comment:
VIRTUAL_ENV could also be `/srv/zulip-venv` or even something else entirely; would prefer to just check it's nonempty, or if mayyybe that `zulip` appears in it (e.g. `echo $VIRTUAL_ENV | grep -q zulip`).